config/file_config.go

```diff
@@ -50,7 +50,8 @@ func parseNodesAndPeers(value interface{}) (nodes map[string]FederationNode, pee
 		for _, item := range nodeList {
 			kv := item.(map[string]interface{})
 
-			if publicKey, err := hex.DecodeString(kv["Key"].(string)); err == nil {
+			var publicKey []byte
+			if publicKey, err = hex.DecodeString(kv["Key"].(string)); err == nil {
```

Comment:
I actually prefer the older way

Reply:
the linter sais since the err is in the declaration of the func then its overshadows ...

Reply:
yeah, because that `err` is not scope to its `if` check. Fix that issue, this issue will disappear